tools/talis/genesis.go

```diff
@@ -146,6 +153,34 @@ func createPayload(ips []Instance, chainID, ppath string, squareSize int, mods .
 	return nil
 }
 
+// mainnetVotingPowers contains the current Celestia mainnet staking distribution for more realistic tests.
+var mainnetVotingPowers []int
+
+func getMainnetStake(index int) int64 {
+	if index < 0 {
+		return 0
+	}
+	if len(mainnetVotingPowers) == 0 {
+		// these figures reflect the exact staking values on 09/07/25.
+		mainnetVotingPowers = []int{
```

Comment:
[question] why not define this at the top of the function?

Reply:
to avoid reinitializing it every time. This gets initialized only once, if the mainnet staking distribution is needed, and gets used afterward.